packages/analytics/src/initialize-ids.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DynamicConfig,
+  Gtag,
+  MinimalDynamicConfig
+} from '@firebase/analytics-types';
+import { GtagCommand, GA_FID_KEY, ORIGIN_KEY } from './constants';
+import { FirebaseInstallations } from '@firebase/installations-types';
+import { fetchDynamicConfigWithRetry } from './get-config';
+import { logger } from './logger';
+import { FirebaseApp } from '@firebase/app-types';
+
+/**
+ * Initialize the analytics instance in gtag.js by calling config command with fid.
+ *
+ * NOTE: We combine analytics initialization and setting fid together because we want fid to be
+ * part of the `page_view` event that's sent during the initialization
+ * @param app Firebase app
+ * @param gtagCore The gtag function that's not wrapped.
+ * @param dynamicConfigPromisesList Array of all dynamic config promises.
+ * @param measurementIdToAppId Maps measurementID to appID.
+ * @param installations FirebaseInstallations instance.
+ *
+ * @returns Measurement ID.
+ */
+export async function initializeIds(
+  app: FirebaseApp,
+  dynamicConfigPromisesList: Array<
+    Promise<DynamicConfig | MinimalDynamicConfig>
+  >,
+  measurementIdToAppId: { [key: string]: string },
+  installations: FirebaseInstallations,
+  gtagCore: Gtag
+): Promise<string> {
+  const dynamicConfigPromise = fetchDynamicConfigWithRetry(app);
+  // Once fetched, map measurementIds to appId, for ease of lookup in wrapped gtag function.
+  dynamicConfigPromise
+    .then(config => {
+      measurementIdToAppId[config.measurementId] = config.appId;
+      if (
+        app.options.measurementId &&
+        config.measurementId !== app.options.measurementId
+      ) {
+        logger.warn(
+          `The measurement ID in the local Firebase config (${app.options.measurementId})` +
+            ` does not match the measurement ID fetched from the server (${config.measurementId}).` +
+            ` To avoid analytics events being sent to the wrong measurement ID, update the` +
```

Comment:
Question - is it that the events would be sent to the "wrong Analytics property", rather than the "wrong measurement ID".  IIUC, the measurement ID is how the Web App is associated with its Analytics property, i.e., the actual entity that logs the events is the "Analytics property".  Can anybody confirm this?

Reply:
It would be the wrong property. I suppose it could also be no property if it's an invalid measurement ID so I guess I was just trying to be very literal and not go any farther than saying it would make the `gtag` call using the wrong measurement ID as a parameter, which could cause it to either be sent to the wrong property or no property at all if it's not a valid measurement ID (the previous expired one?).

I'm not sure if that distinction is important but I guess I imagined people not finding it in any of their properties and then saying they were misled or something.

Reply:
Maybe we can flip this statement and say:

To ensure analytics events are always sent to the correct Analytics property, update....

Reply:
Sounds good! Changed.